gdk/static/config_schema.json

```diff
@@ -138,6 +138,10 @@
                                         "us-gov-east-1"
                                     ]
                                 }
+                            },                                
+                            "s3_upload_file_args": {
```

Comment:
Even though this brings great flexibility it comes at a cost which is that now we are loosing some type safety, `foo` is not a valid args an we are saying it is now (even though it will have no effect on the request being submitted)

Reply:
For security is better to only expose allow values for what we want the customer to allow client to do

Reply:
As mentioned [here](https://github.com/aws-greengrass/aws-greengrass-gdk-cli/pull/99#issuecomment-1430598766), the client operation does this validation. Moreover, we should keep up with S3 supported args and release our `gdk` versions based on that if we couple with it (in schema validation).